src/OpenSage.Game/Logic/Object/ObjectCreationList.cs

```diff
@@ -137,6 +137,7 @@ internal override List<GameObject> Execute(BehaviorUpdateContext context, Vector
             debrisObject.LifeTime = context.LogicFrame + lifeTime;
 
             debrisObject.UpdateTransform(context.GameObject.Translation + Offset, context.GameObject.Rotation);
+            debrisObject.Die(DeathType.Normal);
```

Comment:
`GenericDebris` has no `LifetimeUpdate`, so it needs to be told to die right away it seems?